src/Symfony/Component/Workflow/DefinitionBuilder.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+use Symfony\Component\Workflow\Exception\InvalidArgumentException;
+
+/**
+ * Builds a definition.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class DefinitionBuilder
+{
+    private $places = array();
+    private $transitions = array();
+    private $initialPlace;
+
+    /**
+     * @param string[]     $places
+     * @param Transition[] $transitions
+     */
+    public function __construct(array $places = array(), array $transitions = array())
+    {
+        $this->addPlaces($places);
+        $this->addTransitions($transitions);
+    }
+
+    /**
+     * @return Definition
+     */
+    public function build()
+    {
+        return new Definition($this->places, $this->transitions, $this->initialPlace);
```

Comment:
I would prefer if you put it in a new PR. The current doc PR is already pretty big and I would like to merge it as quick as possible.